app/lib/katello/concerns/base_template_scope_extensions.rb

```diff
@@ -284,14 +290,29 @@ def host_subscription_facet(host)
         host.subscription_facet
       end
 
+      def skip_task?(task)
+        # Skip task that doesn't apply errata
+        input = get_task_input(task)
+        input.blank? || input['host'].blank?
+      end
+
+      def get_task_input(task)
+        @_tasks_input[task.id] ||= if task.label == 'Actions::Katello::Host::Erratum::ApplicableErrataInstall'
+                                     task.execution_plan_action.all_planned_actions(Actions::Katello::Host::Erratum::Install).first.try(:input) || {}
+                                   else
+                                     task.input
+                                   end
+      end
+
       def parse_errata(task)
-        @_tasks_errata_cache[task.id] ||= if task.input['errata'].present?
-                                            # katello agent errata
-                                            task.input['errata']
-                                          else
-                                            # rex errata
-                                            TemplateInvocationInputValue.where(:template_invocation_id => task.template_invocation_id).limit(1).pluck(:value).first.split(',')
-                                          end
+        task_input = get_task_input(task)
+        agent_input = task_input['errata'] || task_input['content']
+        # Pick katello agent errata if present
+        # Otherwise pick rex errata. There are multiple template inputs, such as errata, pre_script and post_script we only need the
+        # errata input here.
+        @_tasks_errata_cache[task.id] ||= agent_input.presence || TemplateInvocationInputValue.joins(:template_input)
```

Comment:
👍🏻  you taught me something  new today.  I wasn't familiar with the `presence` method